### ISSUE TITLE
Let routing options available while calculating route

### ISSUE THREAD

**Issue by serrq** [organicmaps/organicmaps#6809]
![1000100283](https://github.com/organicmaps/organicmaps/assets/113377431/f71803d0-590f-4a37-a89c-7e65c58bbde0)

**Comment by biodranik** [comment]
The current implementation shows routing options only if the route can be different with any of these routing options enabled. Showing them always, and always building the same route in some cases despite changing routing options may not be intuitive for users.

Introducing alternate routes will likely also influence routing options and may be related to their visibility.

**Comment by biodranik** [comment]
@serrq how adding stops is related to route options? You can select any destination and add any stop when the route is built.

**Comment by biodranik** [comment]
I do not understand your specific case about stops and related issue. Can you please record a screen video or describe the exact steps?

Now it is possible to press route to button to start building the route, and then (without waiting for it to finish) to select any additional stop and press "add stop".

Your issue is likely not about adding stops, but about being able to change the current route options without waiting until the route is built, right?

**Comment by biodranik** [comment]
What stops you from adding them? It is already possible. As I wrote: select any destination and press "Add Stop" at the bottom.

**Comment by Misalf-git** [comment]
No problem for me: https://github.com/organicmaps/organicmaps/assets/72605561/c36659ee-b265-4b63-a1d6-095ee19f8aa8

Android
OMaps 2023.11.17-17-Web

**Comment by biodranik** [comment]
No, I don't understand the problem. Instead of waiting at step (3), you can select any other stop and add it to the route. Without waiting.

**Comment by biodranik** [comment]
You can do (without waiting until the route is built):
1. A Route From
2. C Route To
3. B Add Stop

How is adding a stop related to the title?
> Let routing options available while calculating route

**Comment by biodranik** [comment]
> App starts automatically the route building without let me the possibility to add a stop (I have to wait that it finish calculations).

I honestly do not understand your issue. The app allows you to add a stop when it's building a route. It works like on that video posted in the comment above. Did you see it?
https://github.com/organicmaps/organicmaps/assets/72605561/c36659ee-b265-4b63-a1d6-095ee19f8aa8

Can you add a stop in the same way as on the video?

> Because the app won’t let me make additions/changes while calculating something I didn’t ask it to do that way: hence the need to intervene while doing the route calculations. 

Organic Maps ALLOWS you to add a stop when a route is still being built and not finished yet. Please check the video above. You just select any object (while a route is being built) and press Add Stop at the bottom.

> Because basically the app is calculating something for which my request is not yet complete.

For most users, in most cases, the app calculates the complete route. Even if the route is not complete, you can add additional stops at any moment and the route will be recalculated.
